test/core/gprpp/status_helper_test.cc

```diff
@@ -99,6 +99,14 @@ TEST(StatusUtilTest, ToAndFromProto) {
   EXPECT_EQ(s, s2);
 }
 
+TEST(StatusUtilTest, ToAndFromProtoWithUTF8Sanitization) {
+  absl::Status s = absl::CancelledError("UTF-8_\x80_Awesome_\xFF_");
+  upb::Arena arena;
+  google_rpc_Status* msg = internal::StatusToProto(s, arena.ptr());
+  absl::Status s2 = internal::StatusFromProto(msg);
+  EXPECT_EQ(s2.message(), "UTF-8_ _Awesome_ _");
```

Comment:
What if the text has a lot of characters that are not UTF-8?  Couldn't this basically render the message unreadable?

Reply:
Yes it will be replaced with space, maybe not readable but I don't think the original message with code (>= 128) is readable either without a proper encoding.

Reply:
One thing I can do alternatively is to serialize the message field in a payload. In this way, it doesn't need to be sanitized because its type is bytes.

Reply:
I changed this PR to store the message in the payload to workaround this limitation. PTAL.